src/envoy/core.clj

```diff
@@ -112,6 +127,35 @@
        (tools/props->map serializer)
        (get-in (tools/cpath->kpath offset))))
 
+(defn- update-consul
```

Comment:
`update-consul` is too generic `overwrite-with` in this case would be a better function name